src/x/parsers/metric_parsers.go

```diff
@@ -0,0 +1,34 @@
+package parsers
+
+import (
+	"github.com/m3db/m3/src/metrics/metric/id"
+	"github.com/m3db/m3/src/x/serialize"
+	"bytes"
+)
+
+func getMetricTagsIterator() serialize.MetricTagsIterator {
+	it := serialize.NewUncheckedMetricTagsIterator(serialize.NewTagSerializationLimits())
+	return it
+}
+
+// GetMetricIDWithoutLe returns a metric id (with sorted tag pairs) without the le tag
+func GetMetricIDWithoutLe(metricID id.RawID) (id.RawID, bool) {
+	it := getMetricTagsIterator()
```

Comment:
nit: why separate this into another function?

Reply:
will consolidate